src/bvals/cc/bvals_cc_in_one.cpp

```diff
@@ -421,18 +498,18 @@ void SendAndNotify(MeshData<Real> *md) {
 //          Guarantees that buffers for MeshBlocks on the same rank are done, but MPI
 //          communication between ranks may still be in process.
 
-// TODO(pgrete) should probaly be moved to the bvals or interface folders
+// TODO(pgrete) should probably be moved to the bvals or interface folders
 TaskStatus SendBoundaryBuffers(std::shared_ptr<MeshData<Real>> &md) {
   Kokkos::Profiling::pushRegion("Task_SendBoundaryBuffers_MeshData");
 
   auto boundary_info = md->GetSendBuffers();
-  bool cache_is_valid = boundary_info.is_allocated();
+  auto sending_nonzero_flags = md->GetSendingNonzeroFlags();
+  auto alloc_status = ResetSendBuffers(md.get());
 
-  auto buffers_used = ResetSendBuffers(md.get(), cache_is_valid);
-
-  if (!cache_is_valid) {
-    ResetSendBufferBoundaryInfo(md.get(), buffers_used);
+  if (!boundary_info.is_allocated() || (alloc_status != md->GetSendBufAllocStatus())) {
```

Comment:
Ping @Yurlungur and @dholladay00 as you reviewed before:

This change here destroy the caching machinery, doesn't it?
`alloc_status` will always be `!= md->GetSendBufAllocStatus()` as `auto alloc_status = ResetSendBuffers(md.get());` internally creates a new vector on every call, or am I missing sth.

Reply:
Yes it creates a new vector every call... I think we want element-wise comparison of the vectors... i.e., if all the bools in the vector are identical, we're good. I.e., we want
```C++
!std::equal(alloc_status.begin(), alloc_status.end(), md-GetSendBufAllocStatus().begin())
```